repo_resource/common.py

```diff
@@ -253,6 +254,24 @@ def __get_remote_url(self, remote):
     def __get_remote_revision(self, remote):
         return self.__remote_revision[remote]
 
+    def set_insteadOf(self, matrix: dict = None):
+        with redirect_stdout(sys.stderr):
+            if matrix is not None:
```

Comment:
Not needed to test if this is not none.
The code will crash which what we expect.

If you really want to keep this check, please refactor as following:
```python
def set_insteadOf(self, matrix: dict = None):
       if matrix is None:
           return self
           
      with redirect_stdout(sys.stderr):
```

Reply:
We do need this. We need to go on if the insteadOf parameter isn't set. The code is not supposed to crash if empty.
Of course, we could check the insteadOf parameter outside the function but it is safer inside imo.

Reply:
But there is not really a reason for calling `set_insteadOf` when `matrix` is `None`, no?

I'm ok to keep the check, but I'd like to see the early error catching implemented as suggested in the first comment.